test/estimate-gas.test.js

```diff
@@ -0,0 +1,138 @@
+import { expect } from 'chai'
+const init = require('./helpers/init')
+const errors = require('./helpers/errors')
+const web3Utils = require('web3-utils')
+const ethjsABI = require('ethjs-abi')
+const createCsvWriter = require('csv-writer').createObjectCsvWriter
+
+
+contract('Gas Estimates', accounts => {
+    const ctx = {
+      actors: init.getTestActorsContext(accounts),
+      contracts: {
+        token: null,
+        atonomi: null
+      }
+    }
+
+    const csvWriter = createCsvWriter({
+        path: 'output/gasOutput.csv',
+        header: [
+            {id: 'scope', title: 'SCOPE'},
+            {id: 'task', title: 'TASK'},
+            {id: 'gas', title: 'COST IN WEI'},
+        ]
+    })
+    const recordsForCSV = []
+    const tokenDecimals = 18
+    const multiplier = 10 ** tokenDecimals
+    const regFee = 1 * multiplier
+    const actFee = 1 * multiplier
+    const repReward = 1 * multiplier
+    const irnReward = repReward * 0.20
+    const mfgReward = repReward - irnReward
+    const deviceId = 'apple-iphone1'
+    const deviceIdHash = web3Utils.soliditySha3({t: 'bytes32', v: web3.fromAscii(deviceId)})
+    const mfgId = 'APPLE'
+    const deviceType = 'phone'
+    const devicePublicKey = 'somepublickey'
+    const score = 'somescore'
+
+    beforeEach(async () => {
+        ctx.contracts.token = await init.getAtonomiTokenContract(ctx.actors.owner, ctx.actors.releaseAgent)
+        ctx.contracts.atonomi = await init.getAtonomiContract(ctx.actors.owner, ctx.contracts.token.address)
+        await ctx.contracts.token.transfer(ctx.actors.deviceOwner, actFee, {from: ctx.actors.owner})
+        await ctx.contracts.token.transfer(ctx.actors.mfg, regFee + actFee, {from: ctx.actors.owner})
+        await ctx.contracts.atonomi.addNetworkMember(ctx.actors.mfg, false, true, false, mfgId, {from: ctx.actors.owner})
+        await ctx.contracts.atonomi.addNetworkMember(ctx.actors.irnNode, false, false, true, '', {from: ctx.actors.owner})
+    })
+
+    describe('global costs', () => {
+        it('   ^= to Deposit Tokens', async() => {
+            var gas = await ctx.contracts.token.transfer.estimateGas(ctx.actors.alice, regFee, {from: ctx.actors.owner})
```

Comment:
use the `depositTokens` call on the contract to calculate the gas, this is calculating the gas for transfer from owner to alice